tests/test_flows.py

```diff
@@ -1246,14 +1249,25 @@ def test_higher_layer_data_payload_trigger(self):
         mid = env.controller.dev_send_1905(env.agents[0].mac, 0x8018,
                                            tlv(0xA0, 0x04b1, "{0x00 %s}" % payload))
 
-        debug("Confirming higher layer data message was received in the agent")
+        debug("Confirming higher layer data message was received in one of the agent's radios")
 
-        self.check_log(env.agents[0], r"HIGHER_LAYER_DATA_MESSAGE")
+        received_in_radio0, _, _ = self.check_log(env.agents[0].radios[0],
+                                                  r"HIGHER_LAYER_DATA_MESSAGE",
+                                                  fail_when_not_found=False)
+        received_in_radio1, _, _ = self.check_log(env.agents[0].radios[1],
+                                                  r"HIGHER_LAYER_DATA_MESSAGE",
+                                                  fail_when_not_found=False)
+        if (received_in_radio0 == received_in_radio1):
+            self.fail(f"higher layer data message was received and acknowledged by "
+                      f"{int(received_in_radio0) + int(received_in_radio1)} agent's radios, "
+                      f"expected only 1")
 
-        debug("Confirming matching protocol and payload length")
+        received_agent_radio = (env.agents[0].radios[0] if received_in_radio0
+                                else env.agents[0].radios[1])
```

Comment:
do we care for debug purposes on which radio we received the message?
if yes, then need to add debug print regarding this.
@morantr

Reply:
It is used for the next check on the log files. Once we know which son_slave received it, we don't need to check on both of them.